experimentation/src/test/java/com/automattic/android/experimentation/ExPlatTest.kt

```diff
@@ -83,14 +84,28 @@ class ExPlatTest {
 
         exPlat.forceRefresh()
 
-        verify(experimentStore, times(1)).fetchAssignments(eq(platform), any(), anyOrNull())
+        verify(restClient, times(1)).fetchAssignments(eq(platform), any(), anyOrNull())
+    }
+
+    @Test
+    fun `force refresh is successful when cache is fresh `() = runBlockingTest {
```

Comment:
I identify some issues with this tests suite. It's tightly coupled with implementation details of `ExPlat` and favors checks on method execution (all checks are `verify`) rather than actual outcomes.

For this reason, it doesn't really test things. E.g. the whole test suite was passing, even though I forgot to add logic for saving cache. See: 82aa328112475c28471bae45a3e0d4d76fda9558 (passing) but it has no saving logic.

The issue is that, e.g. we have "forceRefresh fetches assignments if cache is fresh", which checks if a `fetch` method was executed on a mock. We don't have verifications that assignments has been indeed really fetched. The, added by me in this PR, test "force refresh is successful when cache is fresh" is also not ideal: it checks if we called method on `cache`, but not that cache has been saved.

----

For all of these reasons I plan to work on improving this test suite, but I want to do this in a separate PR to not shadow changes here.